### DISCUSSION TITLE
Impossibile pairing tra router CC2531 e coordinator CC2538+CC2592

### DISCUSSION THREAD

**Discussion by manny199** [indomus/forum#1243] [GENERALE - Discussioni generiche]
Ciao a tutti,
posseggo il coordinator CC2538 + CC2592 di Giovanni ma non riesco a fare il pairing con una CC2531 flashata con firmware router CC2531_router_2020_09_29.
Il coordinator CC2538 + CC2592 si accoppia regolarmente rilevando tutti gli altri dispositivi ma non il router CC2531 che da GUI zigbee2mqtt, abilitando il join, non viene neppure vista. In qualche caso abilitando e disabilitando il permit join nel log vedo l'IEEE address del router CC2531 ma nella schermata principale di z2m nulla.
Preciso che eseguendo la medesima procedura con coordinator CC2531 e router CC2531 il pairing avviene senza alcun problema.
Qualche idea o consiglio?

**Comment by snakuzzo** [comment]
Innanzi tutto verifica il firmware che hai sui due coordinator CC2538 e CC2531.
Nello specifico verifica se si tratta del firmware `default` e di quello `source routing`. [QUI](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator) trovi le spiegazioni sulle differenze.

Poi considera che dovresti aggiornare il firmware all'ultima versione di metà novembre.

**Reply by manny199** [reply]
Grazie ma il firmware source routing fa diventare la CC2531 router?

**Reply by snakuzzo** [reply]
perdonami...ti ho indicato un link dove vengono spiegate le differenze. Se mi fai questa domanda vuol dire che non hai nemmeno cliccato :D
Leggi sotto la tabellina che troverai e vedi che c'è scritto ;)

**Reply by manny199** [reply]
Quello che mi ha colto in inganno è che in quella pagina si parla di coordinator

**Reply by snakuzzo** [reply]
Infatti è quello di cui parliamo...del cooridinator. Hai appurato che la cc2531 come router funziona...va verificato perché non fa il pairing con la cc2538. Quindi va prima verificato lato coordinator

**Reply by manny199** [reply]
Infatti, quello che tento di fare ma ho qualche difficoltà per il flash della CC2538 + CC2592 via USB. 
Devo eseguire uno script cc2538-bsl.py ma non trovo riferimenti su come verificare se il bootloader del firmware da scaricare sia attivo e dunque quale versione e la sintassi da utilizzare per il flash da cmd di windows.
Ovviamente su windows ho python ...

**Reply by snakuzzo** [reply]
Chiedi a chi ti ha venduto la chiavetta
